### ISSUE TITLE
Literal attribute for data-?

### ISSUE THREAD

**Issue by subversivo58** [erusev/parsedown-extra#99]
Is possible literal attribute?

    $Extra = new ParsedownExtra();
    echo $Extra->text('# Header {data-first=ok}');
    //<h1 data-first="ok">Header</h1>

    // or multiples
    echo $Extra->text('# Header {data-first=ok data-second=throw}');
    //<h1 data-first="ok" data-second="throw">Header</h1>

    // or multiple combinations
    echo $Extra->text('# Header {#show .a .b .c data-first=ok data-second=throw}');
    //<h1 id="show" class="a b c" data-first="ok" data-second="throw">Header</h1>

Sorry my poor example.

**Comment by tovic** [comment]
Possible with my `ParsedownExtraPlugin` extension &rarr; https://github.com/tovic/parsedown-extra-plugin

**Comment by subversivo58** [comment]
@tovic its (plugin) applied any inline elements ?

**Comment by tovic** [comment]
No, see **Markdown Extra** specification. It should be applicable for links, images, headers and code blocks only.

**Comment by subversivo58** [comment]
Hum... i look [this issue](https://github.com/erusev/parsedown/issues/262) and imagining if here possible apply specific rules (data-) for after render document, manipulate elements (with this rules) and your interactions with javascript.

For example: I use javascript to translate the language of an application and to be able to follow the same flow and translate one based documentation markdown files you need more of class and ids, need to use other attributes to be able to manipulate the entries with javascript.

I know the library or plugin aims to be lean and exact but I think that although it is not officially supported by Markdown, if the library or plugin to do the conversion to HTML is common sense to expect that you can include (use) attributes of language HTML.

**Comment by tovic** [comment]
Yep. I can understand. But remember that Markdown is _not_ a replacement for HTML.

**Comment by subversivo58** [comment]
Yes I know. But after converted and rendered is assumed as such.

If the purpose of using Markdown files is to convert them to use in HTML pages, I think the valid to incorporate attributes to the exit.

Of course. If this is a specification of bibliotea or if an implementation is possible without bringing damages the library (or plugins :)

Overall I like this source!
